private/buf/cmd/protoc-gen-buf-breaking/breaking.go

```diff
@@ -43,7 +45,12 @@ const (
 
 // Main is the main.
 func Main() {
-	protoplugin.Main(protoplugin.HandlerFunc(handle))
+	protoplugin.Main(
+		protoplugin.HandlerFunc(handle),
+		protoplugin.WithUnmarshalOptions(proto.UnmarshalOptions{
+			Resolver: (*protoregistry.Types)(nil),
```

Comment:
I don't understand what this is doing and why - needs extensive code comments

Reply:
The code has been made clearer, no more typed nil.